bluepyefe/reader.py

```diff
@@ -199,3 +200,90 @@ def nwb_reader(in_data):
         data = reader.read()
 
     return data
+
+
+def csv_lccr_reader(in_data):
+    """Reader to read .txt (csv_lccr)
+    Args:
+        in_data (dict): of the format
+        {
+            'filepath': "./XXX.txt",
+            'dt': 1
+            'ton': 2000,
+            'toff': 2500,
+            'ljp': 14.0,
+            'amplitudes': [0, 10, 20, 30, 40, 50, 60, 70, 80, 90],
+            'hypamp': -0.2
+            'remove_last_100ms': True,
+            'v_unit': 'mV',
+            't_unit': 'ms',
+            'i_unit': 'nA'
+        }
+    """
+    _check_metadata(
+        in_data,
+        csv_lccr_reader.__name__,
+        ["filepath", "dt", "amplitudes", "v_unit", "t_unit", "i_unit", "ton", "toff", "hypamp"],
+    )
+
+    data = []
+
+    fln = os.path.join(in_data['filepath'])
+    if not os.path.isfile(fln):
+        raise FileNotFoundError(
+            f"Please provide a string with the filename of the CSV file; "
+            f"current path not found: {fln}"
+        )
+
+    dt = in_data['dt']
+    ton = in_data['ton']
+    toff = in_data['toff']
+    amplitudes = in_data['amplitudes']
+    hypamp = in_data['hypamp']
+
+    import csv
+    with open(fln, 'rt') as f:
+        reader = csv.reader(f, delimiter='\t')
+        columns = list(zip(*reader))
+        length = numpy.shape(columns)[1]
+
+        voltage = numpy.array([
+            [
+                float(string) if string not in ["-", ""] else 0
+                for string in column
+            ]
+            for column in columns
+        ])
+        t = numpy.arange(length) * dt
+
+    # Remove last 100 ms if needed
+    if in_data.get('remove_last_100ms', False):
+        slice_end = int(-100. / dt)
+        voltage = voltage[:, :slice_end]
+        t = t[:slice_end]
+
+    for ic, voltage in enumerate(voltage):
+        amp = amplitudes[ic]
+        current = numpy.zeros_like(voltage)
```

Comment:
You should add hypamp for the whole trace, no?

Reply:
I think I set it here:
https://github.com/BlueBrain/BluePyEfe/blob/e0b10c7113f123e24e315c9de9b89b946106eee7/bluepyefe/reader.py#L280

Reply:
I think we still expect to have the 'correct' current passed down, even if BPE later uses the value you pointed to.
Could you try to have current set as hypamp outside stim, and hypamp + amplitude during stim, and see if we have different output?